map/autoware_lanelet2_divider/autoware_lanelet2_divider/xml_tool/xml_tool.py

```diff
@@ -0,0 +1,157 @@
+        divided_way_list = []
+        divided_relation_list = []
+
+        select_tags(divided_map_root, divided_node_list, divided_way_list, divided_relation_list)
+
+        for relation in divided_relation_list:
+            for r in relation.iter("member"):
+                if r.attrib["type"] == "way":
+                    if r.attrib["ref"] not in [way.attrib["id"] for way in divided_way_list]:
+                        for way in way_list:
+                            if way.attrib["id"] == r.attrib["ref"]:
+                                divided_map_root.append(way)
+                        select_tags(
+                            divided_map_root,
+                            divided_node_list,
+                            divided_way_list,
+                            divided_relation_list,
+                        )
+                elif r.attrib["type"] == "relation":
+                    if r.attrib["ref"] not in [rela.attrib["id"] for rela in divided_relation_list]:
+                        for rel in relation_list:
+                            if rel.attrib["id"] == r.attrib["ref"]:
+                                divided_map_root.append(rel)
+                        select_tags(
+                            divided_map_root,
+                            divided_node_list,
+                            divided_way_list,
+                            divided_relation_list,
+                        )
+
+        # Iterate on divided map's ways and find missing nodes
+        for way in divided_way_list:
+            for n in [nd.attrib["ref"] for nd in way.iter("nd")]:
+                if n not in [node.attrib["id"] for node in divided_node_list]:
+                    # find the node in the whole map and add it to the divided map
+                    for node in node_list:
+                        if node.attrib["id"] == n:
+                            divided_map_root.append(node)
+                    select_tags(
+                        divided_map_root,
+                        divided_node_list,
+                        divided_way_list,
+                        divided_relation_list,
+                    )
+
+        divided_map_xml.write(input_extracted_osm_folder + "/" + osm_file)
+
+    return True
+
+
+def complete_missing_version_tag(input_osm_file_path: str):
+    """
+    Add missing version attributes to the root and its child elements in an OSM file, if needed.
+
+    Parameters:
+        input_osm_file_path (str): Path to the OSM file to update with version tags.
+
+    Returns:
+        None
+    """
+    Debug.log(
+        f"Completing missing version tags in osm file: {input_osm_file_path}",
+        DebugMessageType.INFO,
+    )
+
+    whole_map_xml = ET.parse(input_osm_file_path)
+    whole_map_root = whole_map_xml.getroot()
+
+    add_version = any(root_element != "version" for root_element in whole_map_root.attrib)
+
+    if add_version:
+        whole_map_root.set("version", "0.6")
```

Comment:
Why is the version fixed to 0.6?
I’d like to know the reason for choosing 0.6.

Reply:
@ataparlar Do you have any idea on that?

We are adding version tag because osmium tool returns an error if there is missing tag, but I am not sure on the these numbers?

Reply:
@ataparlar Do you have any idea on that?